hudi-client/hudi-client-common/src/main/java/org/apache/hudi/common/table/log/LogFileIterator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.common.table.log;
+
+import org.apache.hudi.common.model.HoodieRecord;
+import org.apache.hudi.common.util.Option;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class LogFileIterator<T> extends CachingIterator<HoodieRecord<T>> {
+  HoodieMergedLogRecordScanner scanner;
+  Map<String, HoodieRecord> records;
+  Iterator<HoodieRecord> iterator;
+
+  protected Option<HoodieRecord> removeLogRecord(String key) {
+    if (records.containsKey(key)) {
+      return Option.of(records.remove(key));
+    }
+    return Option.empty();
```

Comment:
Can be simplified as `return Option.ofNullable(records.remove(key));`

Reply:
cool!